addon/components/o-s-s/side-panel.hbs

```diff
@@ -0,0 +1,28 @@
+<div
+  class="oss-side-panel__container {{if @visible 'oss-side-panel__container--visible'}}"
+  {{will-destroy this.onDestroy}}
+>
+  <div
+    class="oss-side-panel__backdrop {{unless @backdrop 'oss-side-panel__backdrop--transparent'}}"
+    {{on "click" @onClose}}
+    {{on-click-outside this.onClickOutside}}
```

Comment:
That's a good feedback

What we could do it add an argument to the onclose method that is triggered so that the parent can tell if the modal was closed "ungracefully".

Something like `onClose(exitMethod?: 'click-outside' | 'cross-button' | 'escape-key')`

Agreed, this can be done in another PR though